### PR TITLE
Adding round close notification page and relevant validations before directing the page to the user.

### PR DESCRIPTION
**Pull request by nuwan-samarasinghe**
Adding round close notification page and relevant validations before directing the page to the user.

https://dluhcdigital.atlassian.net/browse/FS-4666

### Change description
_A brief description of the pull request_

- [x] Unit tests and other appropriate tests added or updated
- [x] README and other documentation has been updated / added (if needed)
- [x] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
When user is in the middle of a application filling process & have a round close same time will show this page

### Screenshots of UI changes (if applicable)

image

### CODE REVIEW

File: app/default/application_routes.py [hunk 1]
```diff
@@ -135,6 +137,46 @@ def decorator(*args, **kwargs):
...
+        as_dict=False,
+        ttl_hash=get_ttl_hash(Config.LRU_CACHE_TIME),
+    )
+    round_data = get_round_data_without_cache(
+        fund_id=application.fund_id, round_id=application.round_id, language=application.language
+    )
+    deadline = datetime.strptime(round_data.deadline, "%Y-%m-%dT%H:%M:%S")
+    # added this check sometimes if url forcefully change we should not show this notification unless if
+    # there is a round close
+    if datetime.now() > deadline:
```
Comment (srh-sloan) [inline app/default/application_routes.py:157]: Small thing, but frontend already has a way to determine if a round is closed. Would be good to reuse what we have and also if the logic for closing a round ever changed, it's all in one place: https://github.com/communitiesuk/funding-service-design-frontend/blob/aff748a7e17f9390e7e99ea4f0424d2dd063168b/app/default/data.py#L337

**Review by srh-sloan — CHANGES_REQUESTED** [review]
One change but otherwise looks good

File: app/default/application_routes.py (hunk 1 above)
Comment (nuwan-samarasinghe) [reply app/default/application_routes.py]: Fixed

**Review by srh-sloan — APPROVED** [review]
🚀

**Review by srh-sloan — APPROVED** [review]

**Pull request merged**